app/models/authentication.rb

```diff
@@ -44,6 +46,10 @@ def status_severity
     STATUS_SEVERITY[status.to_s.downcase]
   end
 
+  def retryable_status?
+    RETRYABLE_STATUS.include?(status.to_s.downcase)
```

Comment:
wonder if there should be a normalized_status method or something and use it here and 5 lines up (amongst other places)